File: php/features.php
<?php
declare(strict_types=1);

// UvIndexApi2 SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class UvIndexApi2Features
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new UvIndexApi2BaseFeature();
            case "test":
                return new UvIndexApi2TestFeature();
            default:
                return new UvIndexApi2BaseFeature();
        }
    }

    /**
     * Does a generated feature class back this name? False for a name only
     * an options extend instance can supply (the station adopt path) - the
     * constructor uses this to skip make_feature for such names instead of
     * adding a stray BaseFeature.
     */
    public static function has_feature(string $name): bool
    {
        switch ($name) {
            case "base":
            case "test":
                return true;
            default:
                return false;
        }
    }
}

File: py/uvindexapi2_sdk/features.py
# UvIndexApi2 SDK feature factory

from uvindexapi2_sdk.feature.base_feature import UvIndexApi2BaseFeature
from uvindexapi2_sdk.feature.test_feature import UvIndexApi2TestFeature


def _make_feature(name):
    features = {
        "base": lambda: UvIndexApi2BaseFeature(),
        "test": lambda: UvIndexApi2TestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
